### PR TITLE
docs(guide): dataDir init; unix-domain socket length limitation

### PR DESCRIPTION
**Pull request by shivaraj-bh**
Preview:
```sh
nix run "github:juspay/services-flake/guide-datadir?dir=doc"
```

### CODE REVIEW

File: doc/guide/datadir.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# dataDir
+
+The `dataDir` of these services tend to take *relative* paths, which are usually relative to the project root. As such, when you run these services using `nix run`, their data files are created relative to whichever directory you are in. If you want these data files to always reside relative to the project directory, instead of using `nix run` consider wrapping the process-compose packages in script, via either [mission-control](https://community.flake.parts/mission-control) module or a [justfile](https://just.systems/). `services-flake` uses the latter.
```
Comment (srid) [inline doc/guide/datadir.md:3]: I'd expect the very first sentence of a guide/*` to explain *what* `dataDir` is.

File: doc/guide/datadir.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+## Default data directory structure
```
Comment (srid) [inline doc/guide/datadir.md:5]:
```suggestion
{#default-structure}
## Default data directory structure
```

File: doc/guide/datadir.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+```sh
+|-- data
+|   |-- pg
+|   |-- rd
+```
+
+## Gotchas
+
+### Unix-domain socket path is too long
```
Comment (srid) [inline doc/guide/datadir.md:27]:
```suggestion
{#socket-path}
### Unix-domain socket path is too long
```

File: doc/guide/datadir.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+unix socket length is limited to [about 100 chars](https://linux.die.net/man/7/unix). If your data directory is nested too deep, you will have to set `dataDir` option of the service to a shorter path as a workaround.
```
Comment (srid) [inline doc/guide/datadir.md:29]: Should this also talk about `sockerDir`? Or would that be separate PR?

File: doc/guide/datadir.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
+# dataDir
```
Comment (srid) [inline doc/guide/datadir.md:1]:
```suggestion
# Data directory
```

File: doc/guide/datadir.md (hunk 4 above)
Comment (shivaraj-bh) [reply doc/guide/datadir.md:29]: socketDir is part of postgres, so it will be as part of postgres documentation. i will be raising that next

File: doc/guide/datadir.md (hunk 1 above)
Comment (shivaraj-bh) [reply doc/guide/datadir.md:5]: https://github.com/juspay/services-flake/pull/144/commits/37136ca4a0b10050accf0d01eb6d751e20eae146

File: doc/guide.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+---
+short-title: Guide
+---
```
Comment (srid) [inline doc/guide.md:3]: Can be removed, since it is the same as title.

```suggestion
```
